pep-0007.txt

```diff
@@ -31,13 +31,12 @@ particular rule:
 C dialect
 =========
 
-* Python versions before 3.6 use ANSI/ISO standard C (the 1989 version
-  of the standard).  This means (amongst many other things) that all
-  declarations must be at the top of a block (not necessarily at the
-  top of function).
+* Python 3.11 and newer versions use C99 in the public C API and use a
+  subset of C11 in Python internals. The public C API should be
+  compatible with C++. The C11 subset are features supported by GCC 8.5,
+  clang 8.0, and MSVC of Visual Studio 2017.
```

Comment:
The public C API should be compatible with C11 as well.
C11 made several features present in C99 optional (most notably variable-length arrays, MSVC doesn't have). Those shouldn't be used in the API.

Reply:
Ok, let me try something simpler: "Python 3.11 and newer versions use C11 without optional features". Is it better? (see the updated PR)

Reply:
Not sure if you want to add a reference for that, but a simple overview is:
https://en.wikipedia.org/wiki/C11_%28C_standard_revision%29#Optional_features

Reply:
That indeed seems helpful